scripts/component-templates.js

```diff
@@ -0,0 +1,182 @@
+/* eslint-disable prettier/prettier */
+'use strict';
+
+const lowerFirst = require('lodash.lowerfirst');
+const snakeCase = require('lodash.snakecase');
+const toUpper = require('lodash.toupper');
+
+// component.tsx
+const component = (name) => `import styled, { css } from 'styled-components';
+import { FlattenSimpleInterpolation, ThemedStyledProps } from 'styled-components/ts3.6';
+
+import { TTheme } from '../../foundations/theme-provider/themes/theme.types';
+
+import ${name}Base from './${name}.base';
+import {
+    DEFAULT_${toUpper(snakeCase(name))}_SIZE,
+    ${toUpper(snakeCase(name))}_DEFINITIONS,
+} from './${name}.constants';
+import P${name} from './${name}.props';
+
+const ${name} = styled(${name}Base)<P${name}>(
+    ({
+        size = DEFAULT_${toUpper(snakeCase(name))}_SIZE,
```

Comment:
just to keep in mind to follow the naming convention, perhaps we should document that later

Reply:
it does ... and yes ... we should document that.